backend/src/routing/controllers/api/Facharbeit.ts

```diff
@@ -18,4 +19,73 @@ export default class Facharbeit {
         `;
         getAllResults(sql, [], defaultGetAllCallback(res));
     }
+    static POSTchooseTopic(req: Request, res: Response): void {
+        if (rejectWhenValidationsFail(req, res)) return;
+
+        let sql = 'SELECT id FROM facharbeiten WHERE studentID = ?';
+        const studentId = getStudentId();
+        getFirstResult(sql, [studentId], (obj, err) => {
+            if (err) {
+                res.status(500).json(err.name);
+            } else {
+                if (obj) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    updateColumns(obj.id);
+                } else {
+                    insertFirstSubmission();
+                }
+            }
+        });
+
+        const {topic, subject, choosenTeacher, subjectTeacher} = req.body;
+        const args: (string|number)[] = [];
+
+        const buildSetString = (): string => {
+            let setString = '';
+            let first = true;
+            if (topic) {
```

Comment:
Könntest ne Funktion schreiben die 2 argumente nimmt addTosetString(fieldName, stringToAdd) dann machst du first und setString 'global' und dann kannst du in ner for each schleife 4 mal die fkt aufrufen statt hier 4 mal dieses if else statement zu machen 

Kannste aber auch einfach so lassen :D

Reply:
Erledigt.